src/address/address.ts

```diff
@@ -7,11 +7,11 @@ export interface AddressRequestBody {
     lastName: string;
     company: string;
     address1: string;
-    address2: string;
+    address2: string | null;
     city: string;
     stateOrProvince: string;
     stateOrProvinceCode: string;
-    countryCode: string;
+    countryCode: string | null;
```

Comment:
This was my bad, I got confused with `country` field, which shouldn't be passed, but that's the response object.

Currently we seem to be accepting `countryCode: null` but it magically defaults to the US when you do that. I think that could be a bug in our APIs.

Most of the fields are not required, so maybe methods accepting addresses should have `Partial<AddressRequestBody>` as type. Would that solve your problems?

Reply:
Right, this looks like a bug in the Server API. Should always return a string.

Reply:
Will investigate and get back to you.

Reply:
If that's a copy/paste of your code, it seems like it might be broken. 

```
return addresses.map(a => a.countryCode).find(e.length > 0);
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find

`find` expects a function, `e` is `undefined`, yes. I guess you wanted something like: 

```
return addresses
	.map(a => a.countryCode)
	.find(e => e.length > 0);
```

Reply:
FYI, we have identified places in our API where is returning `null` were a string is expected. Thanks for the effort put into this, feel free to close it if you think your issue has been solved. The fix probably won't be live in all stores until early next week.

Reply:
@daisy1754 Keep in mind `customer.addresses` could be an empty array if customer is not signed in.
```customer: {
id: 0,
isGuest: true,
email: "",
firstName: "",
lastName: "",
fullName: "",
addresses: [ ],
storeCredit: 0
}

Reply:
thank you @capsula4